src/components/forms/BookLoanForm.jsx

```diff
@@ -0,0 +1,126 @@
+import React, { useEffect, useState } from "react";
+
+import Button from "@mui/material/Button";
+import TextField from "@mui/material/TextField";
+import Grid from "@mui/material/Grid";
+import Box from "@mui/material/Box";
+import Typography from "@mui/material/Typography";
+import Select from "@mui/material/Select";
+import MenuItem from "@mui/material/MenuItem";
+import FormControl from "@mui/material/FormControl";
+import InputLabel from "@mui/material/InputLabel";
+import { DesktopDatePicker } from "@mui/x-date-pickers/DesktopDatePicker";
+import { LocalizationProvider } from "@mui/x-date-pickers/LocalizationProvider";
+import { AdapterMoment } from "@mui/x-date-pickers/AdapterMoment";
+
+import axios from "axios";
+import { toast } from "react-toastify";
+import moment from "moment";
+
+const BookLoanForm = ({ bookLoan, handleClose }) => {
+  const [status, setStatus] = useState("");
+  const [dateBorrowed, setDateBorrowed] = useState(null);
+  const [dateDue, setDateDue] = useState(null);
+  const [dateReturned, setDateReturned] = useState(null);
+
+  useEffect(() => {
+    setStatus(bookLoan.status);
+    setDateBorrowed(bookLoan.date_borrowed);
+    setDateDue(bookLoan.date_due);
+    setDateReturned(bookLoan.date_returned);
+  }, [bookLoan]);
+
+  const handleStatusChange = (event) => {
+    setStatus(event.target.value);
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    axios
+      .put(`${process.env.REACT_APP_BACKEND_URL}/loans/${bookLoan.id}/`, {
+        book: bookLoan.book.id,
+        status: status,
+        date_borrowed: moment(dateBorrowed).format("YYYY-MM-DD"),
+        date_due:  moment(dateDue).format("YYYY-MM-DD"),
+        date_returned: moment(dateReturned).format("YYYY-MM-DD"),
+      })
+      .then(({ data }) => {
+        toast.success("Book loan updated successfully.");
+      })
+      .catch((err) => {
+        toast.error(err.msg);
+      });
```

Comment:
we can create a util that take request params, urls, etc and call that util here. in this way, we don't have to handle catch everywhere